### ISSUE TITLE
[QST] How to profile the CUTLASS with all of the optimization techniques?

### ISSUE THREAD

**Issue by ybai62868** [NVIDIA/cutlass#1207]
**What is your question?**
Hi, 

Thanks for the great work! 
Recently, I am exploring the performance improvement from all of the optimization in CUTLASS.
I want to profile all of the optimizaiton introduced in CUTLASS and know which optimization is too important or not.
Specificially, as for a GEMM operator, I want to know the performance improvment from which optimization such as datalayout design or thread block swizzle or  tensor core or so on. It is alike the ablation study. 
Can you give me some suggestions to do this exploration?

Thanks,
Yang

**Comment by thakkarV** [comment]
your best bet is to take the SM80 mainloop from CUTLASS 3.x and replace the tiled mma and tiled copies, change pipeline depth etc. you can find a testcase for it here: https://github.com/NVIDIA/cutlass/blob/main/test/unit/gemm/device/sm80_gemm_f16_f16_f32_tensor_op_f32.cu

Note that 3.x Ampere kernel are not optimized to the same extent 2.x API kernel are, but you can only do what you are trying to with the CuTe based implementations.

**Comment by ybai62868** [comment]
Hi, @thakkarV 
Thanks for your quick response! I have already known how to change the tiling size and pipelining stage. 
Is there anything to verify the threadblock swizzle or other performance specific in CUTLASS.
Thanks a lot!

**Comment by thakkarV** [comment]
you can change the thread or the val layouts for the copies to change the coalescing or vectorization extents. you can change the smem layouts to different swizzles or any affine layout you want to change the STG/LDS vectorization or bank conflict count etc tons of things you can do

**Comment by ybai62868** [comment]
@thakkarV 
Hi, thanks for your reply. I have used the template provided in include/cutlass/device/gemm.h to build my custom GEMM kernel to do the profiling but I find a question that when I set the "numStage = 1“
I cannot get the correct results produced by the CUTLASS compared with the host result.
How to solve this problem? In other word, I want to know how the multi-stage pipelining can improve the kernel performance of the CUTLASS. I want to do an ablation study about "w/o multi-stage, w/ multi-stage".

Thanks a lot!
